docs/Deprecated-functionality.rst

```diff
@@ -34,11 +34,43 @@ For authors of easyconfig files:
 
 For developers of easyblocks:
 
-*(nothing yet)*
+* ref:`depr_error_reporting`
 
 For EasyBuild framework developers:
 
-*(nothing yet)*
+* ref:`depr_error_reporting`
+
+.. _depr_error_reporting:
+
+Report errors by raising ``EasyBuildError`` rather than using ``error()``/``exception()`` log methods
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Use of the** ``error`` **and** ``exception`` **log methods is deprecated.**
```

Comment:
We shouldn't use them any more or, we shouldn't use them any more for a fatal error?